### PR TITLE
Github CI to test metrics integration

### PR DESCRIPTION
**Pull request by uSaiPrashanth**
This PR adds github CI to test metrics integration

For now, we trigger it manually. Can be automated if required later...

### CODE REVIEW

File: calculate_metrics.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def parse_cli_args():
         "--datasets",
         type=str,
         help=dataset_args_help,
-        choices=datasets_args_default,
+        choices=datasets_args_default + ["test"],
```
Comment (alvin319) [inline calculate_metrics.py:62]: I think you just add the `test` option to `datasets_args_default`

File: calculate_metrics.py [hunk 2]
```diff
@@ -189,6 +203,8 @@ def load_precomputed_features(schema: str) -> Dict[PrecomputedFeatureName, DataF
             continue
 
         LOGGER.info(f"Downloading dataset {name}-{split}...")
+        if is_test:
+            split = split + "[:3000]"
```
Comment (alvin319) [inline calculate_metrics.py:207]: the string of  `[:3000]` will actually truncate up to the first 3K rows? huggingface's dataset implementation seems very janky

File: calculate_metrics.py [hunk 3]
```diff
@@ -242,7 +258,7 @@ def main():
 
     for data_scheme in args.schemes if isinstance(args.schemes, list) else args.schemes.split(","):
         LOGGER.info("Loading pre-computed features...")
-        precomputed_features = load_precomputed_features(data_scheme)
+        precomputed_features = load_precomputed_features(data_scheme, is_test = str(args.datasets) == "test")
```
Comment (alvin319) [inline calculate_metrics.py:261]: I think we should just put this loop `for dataset_name in args.datasets if isinstance(args.datasets, list) else args.datasets.split(",")` right outside of the `data_scheme` loop, then we can calculate `is_test` directly off of the right `dataset_name`. 

The current implementation would only work if we only run on `test` datasets

**Review by alvin319 — APPROVED** [review]
lgtm - left some small comments

File: calculate_metrics.py (hunk 2 above)
Comment (uSaiPrashanth) [reply calculate_metrics.py:207]: Yes it does! https://huggingface.co/docs/datasets/v1.11.0/splits.html

File: calculate_metrics.py (hunk 1 above)
Comment (uSaiPrashanth) [reply calculate_metrics.py:62]: Changed!

**Pull request merged**